### ISSUE TITLE
Move temp dir to /data

### ISSUE THREAD

**Issue by mmattel** [owncloud/core#12910]
Based on: https://github.com/owncloud/core/issues/5949#issuecomment-67312666

**Background:**
Most of the external storage types do right now not support streaming. This means, that any file transferred needs to be buffered at a temporary space on the server who runs owncloud. This can cause severe space issues, even if the space needed is only for a limited period of time. Right now, the temporary location is defined at https://github.com/owncloud/core/blob/master/lib/base.php#L980
It is either getting the windows version of the tmp dir or via php (sys_get_temp_dir) where unix will put it to.

**Suggestion:**
Define the temp dir used at /data (in eg a subdirectory).

The advantage having the temp dir in /data would be, that the temp dir is independent of the OS and webserver used and "outsourceable". 'datadirectory' in config.php is a configurable item...

We would immediate avoid possible space problems at the server caused by temporary files and owncloud related things stay together and are not distributed around.

If the /data directory stays where it is originally setup, nothing will happen negatively as it uses the same physical location of the owncloud installation (and where eg the php temp dir ist set), but as mentioned above it could be relocated easily.

**Comment by oparoz** [comment]
+1 for having a local tmp store. It would be very useful for previews as well.

**Comment by derkostka** [comment]
+1 for a configurable /tmp Folder, here is why (OC 8.0.3RC1):

I have mounted a SMB Share within Owncloud (libsmbclient) and created a link to a file that is 2,15 GB of size. I looked into my php.ini, set the /tmp to some place where i have got enough free bytes ...

Now, smbclient in OC8 still puts the smb.down files into the /tmp (ubuntu system, 1 GB) storage. The result is that the download breaks at 1 GB, without any warning or something.
Note: i did change php.ini, but it does not seem to work out with libsmbcilient.
